src/roster/mod_roster.erl

```diff
@@ -316,58 +315,42 @@ create_sub_el(Items, Version) ->
             children = Items}].
 
 -spec get_user_roster(Acc, Params, Extra) -> {ok, Acc} when
-    Acc :: mongoose_acc:t(),
-    Params :: #{jid := jid:jid()},
+    Acc :: [roster()],
+    Params :: #{show_full_roster := boolean(), jid := jid:jid()},
     Extra :: gen_hook:extra().
-get_user_roster(Acc, #{jid := #jid{luser = LUser, lserver = LServer}}, #{host_type := HostType}) ->
-    NewAcc = get_user_roster(Acc, LUser, LServer, HostType),
-    {ok, NewAcc}.
-
--spec get_user_roster(mongoose_acc:t(),
-                      jid:luser(),
-                      jid:lserver(),
-                      mongooseim:host_type()) -> mongoose_acc:t().
-get_user_roster(Acc, LUser, LServer, HostType) ->
-    case mongoose_acc:get(roster, show_full_roster, false, Acc) of
-        true ->
-            Roster = get_roster(HostType, LUser, LServer),
-            mongoose_acc:append(roster, items, Roster, Acc);
-        _ ->
-            Roster = lists:filter(fun (#roster{subscription = none, ask = in}) ->
-                                          false;
-                                      (_) ->
-                                          true
-                                  end, get_roster(HostType, LUser, LServer)),
-            mongoose_acc:append(roster, items, Roster, Acc)
-    end.
-
+get_user_roster(Items, #{show_full_roster := Full, jid := JID}, #{host_type := HostType}) ->
+    NewItems = do_get_user_roster(HostType, JID, Full),
+    {ok, Items ++ NewItems}.
+
+-spec do_get_user_roster(mongooseim:host_type(), jid:jid(), boolean()) -> [roster()].
+do_get_user_roster(HostType, #jid{luser = LUser, lserver = LServer}, true) ->
+    get_roster(HostType, LUser, LServer);
+do_get_user_roster(HostType, #jid{luser = LUser, lserver = LServer}, false) ->
+    Roster = get_roster(HostType, LUser, LServer),
+    Fun = fun(#roster{subscription = S, ask = A}) -> not (none =:= S andalso in =:= A) end,
+    lists:filter(Fun, Roster).
```

Comment:
Boolean arguments are sometimes confusing. Later in this module there is a call `do_get_user_roster(mongoose_acc:host_type(Acc), JID, false)` where `false` value tells very little about it. What's more, when reading function heads of `do_get_user_roster` you get little as well - you need to read the implementation to understand what this argument changes. Maybe we could introduce two functions `do_get_user_roster/2` and `do_get_user_full_roster/2` instead and call them conditionally in the hook handler without further passing of the `Full` variable?